internal/integration/api/common.go

```diff
@@ -0,0 +1,199 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//go:build integration_api
+
+package api
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/siderolabs/go-retry/retry"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/kubectl/pkg/scheme"
+
+	"github.com/siderolabs/talos/internal/integration/base"
+	"github.com/siderolabs/talos/pkg/machinery/client"
+)
+
+// CommonSuite verifies some default settings such as ulimits.
+type CommonSuite struct {
+	base.K8sSuite
+
+	ctx       context.Context //nolint:containedctx
+	ctxCancel context.CancelFunc
+}
+
+// SuiteName ...
+func (suite *CommonSuite) SuiteName() string {
+	return "api.CommonSuite"
+}
+
+// SetupTest ...
+func (suite *CommonSuite) SetupTest() {
+	// make sure API calls have timeout
+	suite.ctx, suite.ctxCancel = context.WithTimeout(context.Background(), 1*time.Minute)
+
+	if testing.Short() {
+		suite.T().Skip("skipping in short mode")
+	}
+}
+
+// TearDownTest ...
+func (suite *CommonSuite) TearDownTest() {
+	if suite.ctxCancel != nil {
+		suite.ctxCancel()
+	}
+}
+
+// TestVirtioModulesLoaded verifies that the virtio modules are loaded.
+func (suite *CommonSuite) TestVirtioModulesLoaded() {
+	if suite.Cluster.Provisioner() != "qemu" {
+		suite.T().Skip("skipping as qemu provisioner is not used")
+	}
+
+	expectedVirtIOModules := []string{
+		"virtio_balloon",
+		"virtio_pci",
+		"virtio_pci_legacy_dev",
+		"virtio_pci_modern_dev",
+	}
+
+	node := suite.RandomDiscoveredNodeInternalIP()
+
+	ctx := client.WithNode(suite.ctx, node)
+
+	fileReader, err := suite.Client.Read(ctx, "/proc/modules")
+	defer func() {
+		err = fileReader.Close()
+	}()
+
+	suite.Require().NoError(err)
+
+	scanner := bufio.NewScanner(fileReader)
+
+	var loadedModules []string
+
+	for scanner.Scan() {
+		loadedModules = append(loadedModules, strings.Split(scanner.Text(), " ")[0])
+	}
+	suite.Require().NoError(scanner.Err())
+
+	for _, expectedModule := range expectedVirtIOModules {
+		suite.Require().Contains(loadedModules, expectedModule, "expected module %s to be loaded", expectedModule)
+	}
+}
+
+// TestCommonDefaults verifies that the default ulimits are set.
+func (suite *CommonSuite) TestCommonDefaults() {
+	expectedUlimit := `
+core file size (blocks)         (-c) 0
```

Comment:
are these values stable with different RAM config?

Reply:
yes, works fine locally and in ci